spec/models/invoice_spec.rb

```diff
@@ -28,8 +28,47 @@
     it { is_expected.to validate_presence_of(:organization) }
     it { is_expected.to validate_presence_of(:ends_at) }
     it { is_expected.to validate_presence_of(:currency) }
+    it { is_expected.to validate_presence_of(:customer_name) }
     it { is_expected.to validate_numericality_of(:amount).
       is_greater_than(0).is_less_than_or_equal_to(Dictionaries.money_max) }
     it { is_expected.to validate_inclusion_of(:currency).in_array(Dictionaries.currencies) }
+
+    context 'check customer date range overlaping' do
+      let!(:org)      { create :organization }
+      let!(:customer) { create :customer }
+      let!(:invoice)  { create :invoice, customer_name: customer.name,
+        organization: org, starts_at: Date.today - 10.days, ends_at: Date.today }
+      let(:invoice1) { build :invoice, customer_name: customer.name,
+        organization: org, starts_at: Date.today - 9.days, ends_at: Date.today + 1.days }
+      let(:invoice2) { build :invoice, customer_name: customer.name,
+        organization: org, starts_at: Date.today - 11.days, ends_at: Date.today }
+      let(:invoice3) { build :invoice, customer_name: customer.name,
+        organization: org, starts_at: Date.today - 9.days, ends_at: Date.today }
+      let(:invoice4) { build :invoice, customer_name: customer.name,
+        organization: org, starts_at: Date.today - 11.days, ends_at: Date.today + 1.days }
+      let(:invoice5) { build :invoice, customer_name: customer.name,
+        organization: org, starts_at: Date.today + 2.days, ends_at: Date.today + 10.days }
+
+      it 'Show error on starts_at' do
+        invoice1.valid?
+        expect(invoice1.errors[:starts_at]).to include('overlaps with another record')
```

Comment:
@yunixon please check if we can change this message? I would prefer to see error like `overlaps with another Invoice`, if possible.